descqa/NumberDensityVersusRedshift.py

```diff
@@ -0,0 +1,308 @@
+                    #loop over magnitude cuts in results dict
+                    for key, value in results.items():
+                        self.save_quantities(dtype, value, f_handle, comment=' '.join(((comment or ''), key, value.get(info, ''))))
+
+        if self.first_pass: #turn off validation data plot in summary for remaining catalogs
+            self.first_pass = False
+
+        #save figure
+        plt.savefig(os.path.join(base_output_dir, 'Nvsz_'+filelabel+'.png'))
+        plt.close()
+        return TestResult(0, passed=True)
+
+    def make_subplot(self, meanz, catalog_data, nplot, f, z0, z0err, catalog_color, validation_color, catalog_label, validation_label):
+
+        results = {}
+        results['meanz'] = meanz
+        if nplot%self.ncolumns==0:  #1st column
+            f.set_ylabel('$'+self.yaxis+'$',size=fsize)
+
+        if nplot+1 <= self.nplots-self.ncolumns:  #x scales for last ncol plots only
+            #print "noticks",nplot
+            for axlabel in f.get_xticklabels():
+                axlabel.set_visible(False)
+                    
+                #prevent overlapping yaxis labels
+                f.yaxis.get_major_ticks()[0].label1.set_visible(False)
+        else:
+            f.set_xlabel('$z$',size=fsize)
+            for axlabel in f.get_xticklabels():
+                axlabel.set_visible(True)
+
+        #plot catalog data if available
+        if len(catalog_data)>0:
+            y,binEdges,_ = f.hist(catalog_data,bins=self.zbins,label=catalog_label,color=catalog_color,lw=lw2,normed=self.normed,histtype='step')
+            results['y'] = y
+
+        #plot validation data if available
+        if z0 is not None and z0>0.:
+            ndata = meanz**2*np.exp(-meanz/z0)
+            if self.normed:
+                norm = self.nz_norm(self.zhi,z0)-self.nz_norm(self.zlo,z0)
+                f.plot(meanz,ndata/norm,label=validation_label,ls='--',color=validation_color,lw=lw2)
+                results['fit'] = ndata/norm
+            else:
+                raise ValueError("Only fits to normed plots are implemented so far")
+            if z0err is not None and z0err > 0.:
+                nlo = meanz**2*np.exp(-meanz/(z0-z0err))
+                nhi = meanz**2*np.exp(-meanz/(z0+z0err))
+                if self.normed:
+                    normlo = self.nz_norm(self.zhi,z0-z0err)-self.nz_norm(self.zlo,z0-z0err)
+                    normhi = self.nz_norm(self.zhi,z0+z0err)-self.nz_norm(self.zlo,z0+z0err)                    
+                    f.fill_between(meanz, nlo/normlo, nhi/normhi, alpha=0.3,facecolor=validation_color)
+                    results['fit+'] = nhi/normhi
+                    results['fit-'] = nlo/normlo
+                else:
+                    raise ValueError("Only fits to normed plots are implemented so far")
+
+        self.post_process_plot(f)
```

Comment:
I think `post_process_plot` should be called *only* when the the plot is done and ready to be saved. Should remove this line here.